#149 - Update documentation and other changes

abjonnes opened the pull request:
#149 

- Expands and corrects documentation
- Add CLI option to download/update Bloom filter; download Bloom filter when container starts
- Move redirect URIs to vendor config files

jmandel commented on README.md:
```diff
@@ -1,69 +1,46 @@
-## Build and run in docker
+## S4S reference stack
+
+The test suite is one component of the [Sync for Science reference stack](https://github.com/sync-for-science/reference-stack-docker), which contains additional services such as DSTU2 and STU3 FHIR servers. The `SMART EHR` servers in the vendor list of the test suite refer to the servers that run with the reference stack, so they will be unavailable if you choose to run the test suite independent ...
```
Say "independently (i.e., outside of the full reference stack)" the first time, just because a possible mis-reading of "independently" is "on your own, without our help".

jmandel commented on README.md:
```diff
@@ -1,69 +1,46 @@
+### Develop in Docker
 
-## Installation
+If you want to develop on the app using Docker, mount the repository on the container. Note that the container still has to be restarted to reflect any changes.
```
Shouldn't python code changes be picked up automatically by Flask when `FLASK_DEBUG=1`? (If not, we should mention this in the note, specifically that flask reloading won't work.

abjonnes replied:
Turns out this does work, but changes to the `features` directory does not trigger a flask reload (which is why I didn't think it was working), since that directory is not part of the flask app. Behave seems to reread the test files every time anyway, so everything should be good for development with `FLASK_DEBUG`.

jmandel commented on README.md:
```diff
@@ -1,69 +1,46 @@
+## Bloom filter
 
-To run the suite, run `behave` with `VENDOR` like `smart` or another value from https://github.com/sync-for-science/test-suite/tree/master/config.
+The test suite uses a [Bloom filter](https://en.wikipedia.org/wiki/Bloom_filter) to validate known codes. The filter is built with the `data/build_bf.py` script, but requires a number of external databases to be downloaded first. As a convenience, we maintain a prebuilt filter, which is downloaded every time the container starts.
```
This is a functionality comment rather than a docs comment, but: it would be good to have the test suite look for the file at a known path and download only if absent. That allows people to mount the file from docker, or to build a new docker image that pre-bakes a version of the file in.

jmandel commented on testsuite/views.py:
```diff
@@ -172,3 +154,29 @@ def initdb():
+    bloom_file_status = {}
+    bloom_file = './data/codes.bf'
+
+    try:
```
See comment above; or more generally let's talk through ways to avoid the requirement that a developer be online to use/develop the test suite. (Could even be as simple as keeping the current code and adding an env variable to skip `update_bloom_filter` on startup.)

abjonnes commented:
Addressed all @jmandel 's points. I changed the Bloom filter logic to download the filter on container start up if it does not exist already, but manually calling the `/update_bloom_filter` endpoint or running `flask get_bloom_filter --force` will overwrite an existing filter.
